liquibase-core/src/main/java/liquibase/command/core/LockServiceCommandStep.java

```diff
@@ -0,0 +1,61 @@
+package liquibase.command.core;
+
+import liquibase.Liquibase;
+import liquibase.Scope;
+import liquibase.command.*;
+import liquibase.database.Database;
+import liquibase.exception.LockException;
+import liquibase.lockservice.LockService;
+import liquibase.lockservice.LockServiceFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Internal command step to be used on CommandStep pipeline to create lock services.
+ */
+public class LockServiceCommandStep extends AbstractCommandStep implements CleanUpCommandStep {
+
+    protected static final String[] COMMAND_NAME = {"lockServiceCommandStep"};
+
+    private LockService lockService;
+
+    @Override
+    public List<Class<?>> requiredDependencies() {
+        return Collections.singletonList(Database.class);
+    }
+
+    @Override
+    public List<Class<?>> providedDependencies() {
+        return Collections.singletonList(LockService.class);
+    }
+
+    @Override
+    public void run(CommandResultsBuilder resultsBuilder) throws Exception {
+        CommandScope commandScope = resultsBuilder.getCommandScope();
+        Database database = (Database) commandScope.getDependency(Database.class);
```

Comment:
Can you explain how this Database dependency is getting set up?

Reply:
LockServiceCS  requires Database.class , and DbUrlCS provides Database.class . During command evaluation at CommandFactory, in the same place that we had the getOrder before now we validate those dependencies and the command steps are added to the CommandDefinition pipeline in the correct order. 

So: 
* CommandFactory finds a command like "TagExists".
*  TagExists requires _LockService_, 
* Factory finds who provides it (LockServiceCS) and adds it to the pipeline before TagExistsCs. 
* Now Factory verifies that LockServiceCS requires _Database_
* Factory finds who provides it (DbUrlCS) and adds it to the pipeline before LockService. 
* DBUrlCS does not requires anything else, so it end.

At the end we have a CommandDefinition with a pipeline of:
* **DbUrlCS** (create database) -> **LockServiceCS** (create lockservice and locks) -> **TagExistsCS** (does the tag magic) . 

Now, LockServiceCS and DbUrlCS also implement **CleanUpCommandStep**. After the commands executions, the pipeline is executed in the reverse order and if the step implements CleanUpCS the cleanUp method is executed. So:
* **LockServiceCS** (releases the lock) -> **DbUrlCS** (closes db connection)

Reply:
What happens if multiple classes register themselves as a provider of `LockService`?

Reply:
Right now it is a "findFirst" filter on the array of CommandSteps. And if nothing provides it an exception is thrown.

 But in next commands we'll need to come up with a way to define the priority as it may happen, specially when working with comands like DiffChangelog. Maybe something like the `@order(1) `used by spring?

Reply:
I think you should throw an exception if there are multiple providers found. Until we add an ordering mechanism, randomly selecting one of the providers could cause inconsistent behavior. The exception would also make it very clear that whatever you just developed is not working properly.

Reply:
Makes sense! will do it. I'll also add the @Beta annotations to the methods to advise anyone trying to use it.